arbnode/resourcemanager/resource_management_test.go

```diff
@@ -6,52 +6,57 @@ package resourcemanager
 import (
 	"fmt"
 	"os"
+	"regexp"
 	"testing"
 )
 
-func updateFakeCgroupv1Files(c *cgroupsV1MemoryLimitChecker, limit, usage, inactive int) error {
-	limitFile, err := os.Create(c.limitFile)
+func updateFakeCgroupFiles(c *cgroupsMemoryLimitChecker, limit, usage, inactive int) error {
+	limitFile, err := os.Create(c.files.limitFile)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(limitFile, "%d\n", limit)
-	if err != nil {
+	if _, err = fmt.Fprintf(limitFile, "%d\n", limit); err != nil {
 		return err
 	}
 
-	usageFile, err := os.Create(c.usageFile)
+	usageFile, err := os.Create(c.files.usageFile)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(usageFile, "%d\n", usage)
-	if err != nil {
+	if _, err = fmt.Fprintf(usageFile, "%d\n", usage); err != nil {
 		return err
 	}
 
-	statsFile, err := os.Create(c.statsFile)
+	statsFile, err := os.Create(c.files.statsFile)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(statsFile, `total_cache 1029980160
+	if _, err = fmt.Fprintf(statsFile, `total_cache 1029980160
 total_rss 1016209408
 total_inactive_file %d
 total_active_file 321544192
-`, inactive)
-	if err != nil {
+`, inactive); err != nil {
 		return err
 	}
 	return nil
 }
 
-func TestCgroupsv1MemoryLimit(t *testing.T) {
+func TestCgroupsMemoryLimit(t *testing.T) {
```

Comment:
This test basically has duplicate code for 3 subtests. You can just describe each as a test-case and de-duplicate the code. (Sorry overlooked on the first review).

https://go.dev/blog/subtests

Reply:
Fixed.